src/FlagConfigurationLoader.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Eppo;
+
+use Eppo\DTO\Flag;
+use Eppo\Exception\HttpRequestException;
+use Eppo\Exception\InvalidApiKeyException;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\SimpleCache\InvalidArgumentException;
+
+class FlagConfigurationLoader
+{
+    private UFCParser $parser;
+    public function __construct(private readonly APIRequestWrapper $apiRequestWrapper, private readonly ConfigurationStore $configurationStore)
+    {
+        $this->parser = new UFCParser();
+    }
+
+    /**
+     * @throws ClientExceptionInterface
+     * @throws InvalidArgumentException
+     * @throws HttpRequestException
+     * @throws InvalidApiKeyException
+     */
+    public function getConfiguration(string $flagKey): ?Flag
+    {
+        if ($this->apiRequestWrapper->isUnauthorized) {
+            throw new InvalidApiKeyException();
+        }
+
+        $configuration = $this->configurationStore->getConfiguration($flagKey);
+
+        if (!$configuration) {
+            $configurations = $this->fetchAndStoreConfigurations();
```

Comment:
This client was built on the idea that the server may or may not be polling in the background, so this handles the cases of no calls yet to the get the configuration and the case where a new flag has been added since the first fetch and so is not present locally.
Actually, without the polling, nothing in this SDK ensures that up-to-date config is available so that's a problem.

PHP isn't great with background jobs - if we just start polling in the `EppoClient::Init` method, we block execution. I'll need to do a bit of digging into a good way to manage this.

Reply:
Oof yeah. Perhaps we can keep track of the last time we polled, and only fetch if its been > 30 seconds since the previous one?